app/Http/Controllers/FrontController.php

```diff
@@ -271,10 +271,12 @@ public function showStats($year = null, $month = null) {
         $stats = ControllerLog::aggregateAllControllersByPosAndMonth($year, $month);
         $all_stats = ControllerLog::getAllControllerStats();
 
-        $homec = User::where('visitor', 0)->where('status', 1)->get();
-        $visitc = User::where('visitor', 1)->where('status', 1)->get();
-        $agreevisitc = User::where('visitor', 1)->where('visitor_from', 'ZHU')->orWhere('visitor_from', 'ZJX')->where('status', 1)->get();
-       
+        $hcontrollers_public = User::where('visitor', '0')->where('status', '1')->where('name_privacy', '0')->orderBy('lname', 'ASC')->get();
+        $hcontrollers_private = User::where('visitor', '0')->where('status', '1')->where('name_privacy', '1')->orderBy('id', 'ASC')->get();
+        $homec = $hcontrollers_public->merge($hcontrollers_private);
+        $vcontrollers_public = User::where('visitor', '1')->where('status', '1')->where('name_privacy', '0')->orderBy('lname', 'ASC')->get();
+        $vcontrollers_private = User::where('visitor', '1')->where('status', '1')->where('name_privacy', '1')->orderBy('id', 'ASC')->get();
+        $visitc = $vcontrollers_public->merge($vcontrollers_private);
```

Comment:
Could we still order by last name even though some names are hidden? Just because my last name comes before yours doesn't reveal that my last name is "Cowan" and yours is "Porter"

Reply:
@iccowan reference #532, this methodology of further obscuring last names was specifically requested by members. I think we need to make the stats page match the look of the roster page so they both display in the same order.

Reply:
Got it okay then it all looks good to me